### PR TITLE
Bump pipeline and git dependencies

### PR DESCRIPTION
**Pull request by i386**
# Description

@michaelneale @vivek a number of new pipeline updates are available we should integrate and test with.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
Still isn't happy with it - probably something else is pulling in 2.10.. perhaps hold off until workflow-api has been fixed?

File: pom.xml [hunk 1]
```diff
@@ -302,6 +302,11 @@
             <artifactId>github-organization-folder</artifactId>
             <version>1.6</version>
         </dependency>
+        <dependency>
```
Comment (michaelneale) [inline pom.xml]: Is there a reason this one is explicitly set?

**Comment by michaelneale** [comment]
passes ATH https://ci.blueocean.io/job/ATH-Jenkinsfile/job/master/1718/

File: blueocean-pipeline-api-impl/pom.xml [hunk 2]
```diff
@@ -109,7 +109,10 @@
             <artifactId>unirest-java</artifactId>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
```
Comment (michaelneale) [inline blueocean-pipeline-api-impl/pom.xml]: @i386 any reason this is here and not test scoped?

Comment (i386) [inline blueocean-pipeline-api-impl/pom.xml]: Scopes are inherited

Comment (michaelneale) [inline blueocean-pipeline-api-impl/pom.xml]: nice.

File: pom.xml [hunk 3]
```diff
@@ -230,27 +230,27 @@
...
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-cps</artifactId>
-            <version>2.25</version>
+            <version>2.28</version>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
-            <version>2.7</version>
+            <version>2.12-alpha-2</version>
```
Comment (svanoort) [inline pom.xml]: @i386 We're ready to take this up to 2.12 now

**Comment by michaelneale** [comment]
@i386 I can't tell what your latest changes are - but they have broken both unit tests and the ATH

File: pom.xml [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
   <properties>
     <java.level>7</java.level>
-    <jackson.version>2.2.3</jackson.version>
+    <jackson.version>2.8.7</jackson.version>
```
Comment (michaelneale) [inline pom.xml]: why?

**Comment by michaelneale** [comment]
:bug: some failures and a lot of dependency changes that need a bit more description/work.

**Comment by michaelneale** [comment]
I don't think we have the luxury of bumping deps all at once unless we have time to baby sit all the tests/ATH. wouldn't it make sense to focus on workflow-api on its own?

File: blueocean-pipeline-api-impl/pom.xml (hunk 2 above)
Comment (michaelneale) [inline blueocean-pipeline-api-impl/pom.xml]: so - why is this dependency here? It wasn't used before right?

**Comment by michaelneale** [comment]
@i386 did you squash commits here ? as I Can't see changes between today and yesterday to know what caused breakage (best to not squash when pushing to a branch for this reason).

**Comment by michaelneale** [comment]
:bee: LGTM

**Comment by jglick** [comment]
Revert 45aa4f0da66b7439023464bb960d0fcca7a8e139 and fix your tests to `checkout scm`.

**Comment by michaelneale** [comment]
@jglick ?? commenting on long closed PRs?

**Comment by jglick** [comment]
> commenting on long closed PRs?

Two days! Well within the range of my review backlog, unfortunately…

> you mean that changing tests to not use git but use checkout scm

No, IIUC the problem is that there are tests which create a `Jenkinsfile` which is just, say,

```groovy
echo 'hi!'
```

and as of 2.13 (using the now-default lightweight mode—controllable via system property) successive builds of that branch project will not display any SCM changelog, because Jenkins is not running all the machinery of  `SCM.checkout` and its changelog parsing—it is just grabbing the `Jenkinsfile` straight out of the selected revision (in the case of GitHub, via API). So the test should actually do something with the source code:

```groovy
node {
  checkout scm
  echo 'hi!'
}
```

Now I *might* try to implement changelog generation even in lightweight mode—in principle the `SCMFileSystem` APIs make it possible, though AFAIK this code path is untested. But it would be a fair amount of work for what seems like a corner case so it is not exactly a high priority.

**Comment by michaelneale** [comment]
@jglick haha yes 2 days - ancient history!

OK I will take a look at this today, seems pretty simple. If that works, makes sense. They are contrived cases. They are only "echo" so they would in theory work on windows (wishful thinking there though).

**Comment by jglick** [comment]
> they would in theory work on windows

`checkout scm` should be safe on Windows too. But rather than guess, why not patch your own `Jenkinsfile` to run your tests on the label `windows`, which ci.jenkins.io supports?

**Comment by michaelneale** [comment]
@jglick we aren't yet running on ci.jenkins.io - but a dogfood instance, but otherwise yes. What we are planning is some windows + browser permutations first before building js on windows (I believe it can work though, but its more of a pain than it should be with npm).

**Comment by vivek** [comment]
@jglick yeah, we will fix the test to do that and upgrade to workflow-multibranch 2.13. thanks.

**Pull request merged**